src/main/java/hudson/tasks/MimeMessageBuilder.java

```diff
@@ -0,0 +1,198 @@
+ */
+public class MimeMessageBuilder {
+
+    public static final String X_JENKINS_JOB = "X-Jenkins-Job";
+    public static final String X_JENKINS_RESULT = "X-Jenkins-Result";
+
+    private String charset = "UTF-8";
+    private String mimeType = "text/plain";
+    private TaskListener listener;
+    private String defaultSuffix;
+    private String from;
+    private String replyTo;
+    private String recipients;
+    private Run<?, ?> run;
+
+    public MimeMessageBuilder() {
+        if (Jenkins.getInstance() != null) {
+            defaultSuffix = Mailer.descriptor().getDefaultSuffix();
+            from = JenkinsLocationConfiguration.get().getAdminAddress();
+            replyTo = Mailer.descriptor().getReplyToAddress();
+        }
+    }
+
+    public MimeMessageBuilder setCharset(String charset) {
+        assert charset != null;
+        this.charset = charset;
+        return this;
+    }
+
+    public MimeMessageBuilder setMimeType(String mimeType) {
+        assert mimeType != null;
+        this.mimeType = mimeType;
+        return this;
+    }
+
+    public MimeMessageBuilder setListener(TaskListener listener) {
+        assert listener != null;
+        this.listener = listener;
+        return this;
+    }
+
+    public MimeMessageBuilder setRun(Run<?, ?> run) {
+        assert run != null;
+        this.run = run;
+        return this;
+    }
+
+    public MimeMessageBuilder setDefaultSuffix(String defaultSuffix) {
+        assert defaultSuffix != null;
+        this.defaultSuffix = defaultSuffix;
+        return this;
+    }
+
+    public MimeMessageBuilder setFrom(String from) {
+        assert from != null;
+        this.from = from;
+        return this;
+    }
+
+    public MimeMessageBuilder setReplyTo(String replyTo) {
+        assert replyTo != null;
+        this.replyTo = replyTo;
+        return this;
+    }
+
+    public MimeMessageBuilder setRecipients(String recipients) {
+        assert recipients != null;
+        this.recipients = recipients;
+        return this;
+    }
+
+    /**
+     * Build a {@link MimeMessage} instance from the set of supplied parameters.
+     * @return The {@link MimeMessage} instance;
+     * @throws MessagingException
+     * @throws UnsupportedEncodingException
+     */
+    public MimeMessage buildMimeMessage() throws MessagingException, UnsupportedEncodingException {
+        MimeMessage msg = new MimeMessage(Mailer.descriptor().createSession());
+
+        msg.setContent("", mimeType);
+        if (StringUtils.isNotBlank(from)) {
+            msg.setFrom(toNormalizedAddress(from));
+        }
+        msg.setSentDate(new Date());
+
+        addRunHeaders(msg);
+        addRecipients(msg);
+
+        if (StringUtils.isNotBlank(replyTo)) {
+            msg.setReplyTo(new Address[]{toNormalizedAddress(replyTo)});
+        }
+        return msg;
+    }
+
+    public InternetAddress toNormalizedAddress(String address) throws UnsupportedEncodingException {
```

Comment:
Just that there is no real use case for a user of the API to call this method directly, since the builder already calls it for the `To`, `From`, and `Reply-To` people. Since it could be made `private`, might as well do so and keep the API simpler.